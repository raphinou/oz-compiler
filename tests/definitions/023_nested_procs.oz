% A new local variables is used by P *before* it is also used by a nested proc definition.
% We need to be sure that the new local in T references the new local in P.
local
   A P B
in
   proc {P V}
      T
   in
      {Show A}   % use before a nested proc request a new local creation for it
      proc {T U}
        A=44
      in
        {Show A}
        {Show U}
        {Show V}
      end
      {T A}
   end
   A = 5
   B = 7
   {P A}
   {For 1 B 1 P}
end
