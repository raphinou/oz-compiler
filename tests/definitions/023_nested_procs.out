5
44
5
5
5
44
5
1
5
44
5
2
5
44
5
3
5
44
5
4
5
44
5
5
5
44
5
6
5
44
5
7
